internal/factsengine/gatherers/saptune.go

```diff
@@ -0,0 +1,128 @@
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	SaptuneGathererName = "saptune"
+)
+
+// nolint:gochecknoglobals
+var whitelistedArguments = map[string]string{
+	"status":          "status --non-compliance-check",
+	"solution-verify": "solution verify",
+	"solution-list":   "solution list",
+	"note-verify":     "note verify",
+	"note-list":       "note list",
+}
+
+// nolint:gochecknoglobals
+var (
+	SaptuneVersionUnsupported = entities.FactGatheringError{
+		Type:    "saptune-version-not-supported",
+		Message: "currently installed version of saptune is not supported",
+	}
+
+	SaptuneUnknownArgument = entities.FactGatheringError{
+		Type:    "saptune-unknown-error",
+		Message: "the requested argument is not currently supported",
+	}
+
+	SaptuneMissingArgument = entities.FactGatheringError{
+		Type:    "saptune-missing-argument",
+		Message: "missing required argument",
+	}
+
+	SaptuneCommandError = entities.FactGatheringError{
+		Type:    "saptune-cmd-error",
+		Message: "error executing saptune command",
+	}
+
+)
+
+type SaptuneGatherer struct {
+	executor utils.CommandExecutor
+}
+
+func NewDefaultSaptuneGatherer() *SaptuneGatherer {
+	return NewSaptuneGatherer(utils.Executor{})
+}
+
+func NewSaptuneGatherer(executor utils.CommandExecutor) *SaptuneGatherer {
+	return &SaptuneGatherer{
+		executor: executor,
+	}
+}
+
+func parseJSONToFactValue(jsonStr string) (entities.FactValue, error) {
+	// Unmarshal the JSON into an interface{} type.
+	var jsonData interface{}
+	if err := json.Unmarshal([]byte(jsonStr), &jsonData); err != nil {
+		return nil, err
+	}
+
+	// Convert the parsed jsonData into a FactValue using NewFactValue.
+	return entities.NewFactValueMod(jsonData)
+}
+
+func (s *SaptuneGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting %s facts gathering process", SaptuneGathererName)
+	saptuneRetriever, _ := saptune.NewSaptune(s.executor)
+	for _, factReq := range factsRequests {
+			var fact entities.Fact
+		if len(factReq.Argument) == 0 {
+			log.Error(SaptuneMissingArgument.Message)
+			fact = entities.NewFactGatheredWithError(factReq, &SaptuneMissingArgument)
+		} else if factValue, err := handleArgument(&saptuneRetriever, factReq.Argument); err != nil {
+			fact = entities.NewFactGatheredWithError(factReq, err)
+		} else {
+			fact = entities.NewFactGatheredWithRequest(factReq, factValue)
+		}
+		facts = append(facts, fact)
+	}
+
+	log.Infof("Requested %s facts gathered", SaptuneGathererName)
+	return facts, nil
+}
+
+func handleArgument(
+	saptuneRetriever *saptune.Saptune,
+	argument string,
+) (entities.FactValue, *entities.FactGatheringError) { 
+	internalArguments, ok := whitelistedArguments[argument]
+
+	if !ok {
+		gatheringError := SaptuneUnknownArgument.Wrap(internalArguments)
+		log.Error(gatheringError)
+		return nil, gatheringError
+	}
+
+	argList := strings.Split(internalArguments, " ")
```

Comment:
Why this split? you already have the command you want to run in the map you created for the whitelist

Reply:
Because when the arguments are passed to the executor they need to be split as an array. Calling `s.executor.Exec("saptune", "some space separated argument)` wouldn't work, right? it should be `("saptune", "some", "space", "separated", "argument")`

Reply:
You could try to store the command values as string in the whitelisted variable